src/main.rs

```diff
@@ -99,28 +128,52 @@ fn main() -> Result<()> {
         info!("Battery status: {:?}", &batteries);
 
         let global = system::get_global_battery(&batteries);
+        let battery_state_name = system::battery_state_to_name(global.state);
+
+        if battery_state_name != last_battery_state_name {
+            last_battery_state_name = battery_state_name.clone();
+
+            match battery_state_name.as_str() {
```

Comment:
Any time you write something like this, what you wanted was an enum :-) Among other things:

1. Enums ensure all possible states are predefined, and the compiler enforces exhaustive matching. In this case, for example, mistyped states in string matching (e.g., `"Chargin"`) would not be caught at compile time.
2. Adding new states or renaming existing ones is safer and more straightforward with enums, as the compiler ensures all usage is updated and is exhaustive.